operator-framework-core/src/main/java/io/javaoperatorsdk/operator/processing/dependent/kubernetes/KubernetesDependentConverter.java

```diff
@@ -16,49 +22,128 @@
 import static io.javaoperatorsdk.operator.processing.dependent.kubernetes.KubernetesDependentResourceConfig.DEFAULT_CREATE_RESOURCE_ONLY_IF_NOT_EXISTING_WITH_SSA;
 
 public class KubernetesDependentConverter<R extends HasMetadata, P extends HasMetadata> implements
-    ConfigurationConverter<KubernetesDependent, KubernetesDependentResourceConfig<R>, KubernetesDependentResource<R, P>> {
+    ConfigurationConverter<KubernetesDependent, KubernetesDependentResourceConfig<R>> {
 
   @Override
   @SuppressWarnings({"unchecked", "rawtypes"})
   public KubernetesDependentResourceConfig<R> configFrom(KubernetesDependent configAnnotation,
-      ControllerConfiguration<?> parentConfiguration,
-      Class<KubernetesDependentResource<R, P>> originatingClass) {
-    var namespaces = parentConfiguration.getNamespaces();
-    var configuredNS = false;
-    String labelSelector = null;
+      ControllerConfiguration<?> controllerConfig,
+      DependentResourceSpec<?, ?> spec) {
+
     var createResourceOnlyIfNotExistingWithSSA =
         DEFAULT_CREATE_RESOURCE_ONLY_IF_NOT_EXISTING_WITH_SSA;
-    OnAddFilter<? extends HasMetadata> onAddFilter = null;
-    OnUpdateFilter<? extends HasMetadata> onUpdateFilter = null;
-    OnDeleteFilter<? extends HasMetadata> onDeleteFilter = null;
-    GenericFilter<? extends HasMetadata> genericFilter = null;
-    Boolean useSSA = null;
+
+    BooleanWithUndefined useSSA = BooleanWithUndefined.UNDEFINED;
+
     if (configAnnotation != null) {
-      if (!Arrays.equals(KubernetesDependent.DEFAULT_NAMESPACES, configAnnotation.namespaces())) {
-        namespaces = Set.of(configAnnotation.namespaces());
-        configuredNS = true;
+      createResourceOnlyIfNotExistingWithSSA =
+          configAnnotation.createResourceOnlyIfNotExistingWithSSA();
+      useSSA = configAnnotation.useSSA();
+    }
+
+    var informerConfiguration = createInformerConfiguration(configAnnotation,
+        controllerConfig,
+        spec);
+
+    return new KubernetesDependentResourceConfig(useSSA, createResourceOnlyIfNotExistingWithSSA,
+        informerConfiguration);
+
+  }
+
+  @SuppressWarnings({"unchecked", "rawtypes"})
+  private InformerConfiguration<R> createInformerConfiguration(KubernetesDependent configAnnotation,
+      ControllerConfiguration<?> controllerConfig,
+      DependentResourceSpec<?, ?> spec) {
+    try {
+      Class<? extends KubernetesDependentResource<?, ?>> originatingClass =
+          (Class<? extends KubernetesDependentResource<?, ?>>) spec.getDependentResourceClass();
+      var dependentInstance = originatingClass.getConstructor().newInstance();
```

Comment:
This isn't going to work, at least, not as-is, as the whole point of having the spec is to delay instantiation of the dependent until much later when the workflow is resolved using the configured `DependentResourceFactory`.

Reply:
This is just to use some default impl methods. WIll check if I can make this happen without that.